src/backend/distributed/transaction/transaction_management.c

```diff
@@ -317,12 +319,24 @@ CoordinatedTransactionCallback(XactEvent event, void *arg)
 			MemoryContext previousContext =
 				MemoryContextSwitchTo(CitusXactCallbackContext);
 
-			if (CurrentCoordinatedTransactionState == COORD_TRANS_PREPARED)
+			if (CurrentCoordinatedTransactionState == COORD_TRANS_PREPARED &&
+				!IsManagementCommand)
 			{
 				/* handles both already prepared and open transactions */
 				CoordinatedRemoteTransactionsCommit();
 			}
 
+			if (!isCitusManagementDatabase())
```

Comment:
I wonder if this should be changed to:

if (CurrentCoordinatedTransactionState == COORD_TRANS_PREPARED &&
    IsManagementCommand && !isCitusManagementDatabase())

Reply:
And also wondering should we have a way to distinguish pool databases from regular citus databases here. I mean, should we also check for CitusHasBeenLoaded() here? It should be a cheap function call given that https://github.com/citusdata/citus/pull/7123 is merged already.

Reply:
citus.management_database_name means citus.control_db in the context of https://github.com/citusdata/citus/pull/7254.

Reply:
Lets unify our naming. @JelteF, @halilozanakgul which name do you think we sould use all across?

-    Control DB
-    Admin DB
-    Management DB

Reply:
other option is:
- Main DB